src/Workspaces/CSharpTest/Formatting/FormattingTests.cs

```diff
@@ -8653,5 +8653,115 @@ public void M()
 }";
             await AssertFormatAsync(code, code);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public async Task FormatIndexExpression()
+        {
+            await AssertFormatAsync(@"
+class C
+{
+    void M()
+    {
+        object x = ^1;
+        object y = ^1
+    }
+}", @"
+class C
+{
+    void M()
+    {
+        object x = ^1;
+        object y = ^1
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public async Task FormatRangeExpression_NoOperands()
+        {
+            await AssertFormatAsync(@"
+class C
+{
+    void M()
+    {
+        object x = ..;
+        object y = ..
+    }
+}", @"
+class C
+{
+    void M()
+    {
+        object x = ..;
+        object y = ..
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public async Task FormatRangeExpression_RightOperand()
+        {
+            await AssertFormatAsync(@"
+class C
+{
+    void M()
+    {
+        object x = ..1;
+        object y = ..1
+    }
+}", @"
+class C
+{
+    void M()
+    {
+        object x = ..1;
+        object y = ..1
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public async Task FormatRangeExpression_LeftOperand()
+        {
+            await AssertFormatAsync(@"
+class C
+{
+    void M()
+    {
+        object x = 1..;
+        object y = 1..
```

Comment:
i just realized.  have we done the work to ensure that ranges don't interfere with completion? i.e. if i want a range from foo to bar, and i type ```foo..``, then the first ```.``` will bring up completion, and hte second will commit any preselected items.  This is an important muscle memory case as people do hits dots in succession for common patterns that the MRu has memorized.  How will we prevent that breaking ranges?

Reply:
I originally asked @OmarTawfik the same thing @CyrusNajmabadi. :smile:

I agree that adding a test or two to make sure this works as expected (and continues to work!) is important. This looks like the right place to add such tests: https://github.com/dotnet/roslyn/blob/master/src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb.

Reply:
NOoe: this should not just be for numbers (which are already handled specially in intellisense).  It also needs to be for anything that could be in a range.

This also concerns me a bit because it depends on none of *clients* ever selecting 'hard select' in the completion API, and then wanting their value ot be usable in a range.  

I think this needs some design thoughts on what's ok here.  The current syntax is depending *heavily* on the current intellisense impl and the in-box completoin providers we've shipped (and have continually tweaked over the years).

Note: The syntax of ```[a, b]``` and ```[a, b)``` does not seem to have this issue i don't believe...

Reply:
@DustinCampbell @CyrusNajmabadi I've taken a note, and will address it in a following PR to the `features/ranges-v2` branch. Thanks a lot for the pointers!